README.md

```diff
@@ -6,94 +6,68 @@ drandjs is a Javascript library able to communicate with a public [drand](https:
 
-The **distkey** argument represents the distributed key of the participants created during the setup phase.
-The key should be in hexadecimal format and should be given out-of-bands (it can be obtained from the group configuration file or fetched from a drand node operator).
-It should look like:
-```
-distkey = "017f2254bc09a999661f92457122613adb773a6b7c74333a59bde7dd552a7eac2a79263bb6fb1f3840218f3181218b952e2af35be09edaee66566b458c92609f7571e8bb519c9109055b84f392c9e84f5bb828f988ce0423ce708be1dcf808d9cc63a610352b504115ee38bc23dd259e88a5d1221d53e45c9520be9b601fb4f578"
-```
 
-Nevertheless, drandjs allows to set this field to "", in which case, drandjs fetches the distributed key *as well as* the randomness, in order to verify the latter.
-Note that in this mode of operation, the server may lie about the distributed key and create any valid randomness it wants.
+### API 
 
-### Randomness
+Here is a list of public function drandjs exposes. Note that all byte-like arguments (such as the randomness), unless otherwise noted, are hexadecimal-encoded strings. 
 
-- `fetchPublic(identity)` fetches the latest *public* randomness at the specified drand node. The returned JSON matches the structure that you can find at the endpoint `address/api/public`.
-- `verify_drand(previous, randomness, round, distkey)` returns `true` if the verification of the given `randomness` against the other parameters was successful, and `false` if an error occured during the verification process.
+- `fetchPublic(identity)` fetches the latest *public* randomness at the specified drand node.  See the [service definition](https://github.com/dedis/drand/blob/master/protobuf/drand/public.proto#L16) for the structure of the resulting JSON.
 
-- `fetchAndVerify(identity, distkey)` sequentially calls those two functions and returns a Promise, representing the eventual completion (or failure) of the verification of the fetched randomness, with the following structure:
+- `verifyDrand(previous, randomness, round, distkey)` returns `true` if the verification of the given `randomness` against the `distkey` over the message formed from both the `previous` and `round` arguments is successful. It returns `false` if an error occured during the verification process. 
+
+- `fetchAndVerify(identity, distkey)` sequentially calls `fetchPublic` and `verifyDrand` and returns a Promise, holding the eventual randomness in case of sucess and the error in case of failure. It returns a JSON structure such as:
```

Comment:
sucess -> success 
"holding the eventual randomness in case of success and the error in case of failure." I am not a fan of this, as both cases return the randomness, but as a resolve or reject.

Reply:
Yes but come to think of it, it should be really the error that is returned and not the randomness in case of an error. What do you think ? Could you change the return promise to get an error ?

Reply:
In case of unverified randomness, we may still want to print the randomness but with a warning

Reply:
Also, the errors are not very user-friendly (i.e. "wrong buffer size for G2"), but I could certainly improve that

Reply:
> we may still want to print the randomness
On the contrary, cryptographic APIs should have safe defaults, and when fetching randomness which is NOT valid, then it should NOT be used. We don't want the user to think it's still somewhat usable. To me it's OK to have an complicated error if you can embed it with a context. For example, "error verifying the signature: <kyberjs error>" => "error verifying the signature: wrong buffer size for G2")